server/game/insightkeyword.js

```diff
@@ -16,9 +16,11 @@ class InsightKeyword extends BaseAbility {
 
     executeHandler(context) {
         let {game, challenge, source} = context;
-        let drawn = challenge.winner.drawCardsToHand(1);
-        game.raiseEvent('onInsight', { challenge: challenge, source: source, card: drawn });
-        game.addMessage('{0} draws a card from Insight on {1}', challenge.winner, source);
+        if(challenge.winner.canDraw()) {
```

Comment:
This is probably nitpicky, but the `canDraw` check should really go in `meetsRequirements`. If there's ever a weird card that allows cancellation of Insight (e.g. Fear Cuts vs Stealth or Intimidate), it shouldn't fire unless it's actually possible for Insight to initiate.

Reply:
Done.